### ISSUE TITLE
Merge_sinks test fail in parallel on Pleiades

### ISSUE THREAD

**Issue by pgrete** [enzo-project/enzo-e#191]
Full error output:
```
ctest -V -R merge_sinks_stationary_parallel
UpdateCTestConfiguration  from :/nobackup/pgrete/src/enzo-e/build-icc-mpi-latest-boost/DartConfiguration.tcl
Parse Config file:/nobackup/pgrete/src/enzo-e/build-icc-mpi-latest-boost/DartConfiguration.tcl
UpdateCTestConfiguration  from :/nobackup/pgrete/src/enzo-e/build-icc-mpi-latest-boost/DartConfiguration.tcl
Parse Config file:/nobackup/pgrete/src/enzo-e/build-icc-mpi-latest-boost/DartConfiguration.tcl
Test project /nobackup/pgrete/src/enzo-e/build-icc-mpi-latest-boost
Constructing a list of tests
Done constructing a list of tests
Updating test list for fixtures
Added 0 tests to meet fixture requirements
Checking test dependency graph...
Checking test dependency graph end
test 47
    Start 47: merge_sinks_stationary_parallel

47: Test command: /nobackup/pgrete/src/ytenv/bin/python3 "input/merge_sinks/run_merge_sinks_test.py" "--launch_cmd" "mpiexec -np 4 /nobackup/pgrete/src/enzo-e/build-icc-mpi-latest-boost/bin/enzo-e" "--prec=double" "--ics_type=stationary"
47: Test timeout computed to be: 1500
47: Charm++> Running on MPI version: 3.1
47: Charm++> level of thread support used: -1 (desired: 0)
47: Charm++> Running in non-SMP mode: 4 processes (PEs)
47: Converse/Charm++ Commit ID: v7.0.0
47: Charm++> MPI timer is synchronized
47: Isomalloc> Synchronized global address space.
47: CharmLB> Load balancer assumes all CPUs are same.
47: Charm++> Running on 1 hosts (2 sockets x 14 cores x 2 PUs = 56-way SMP)
47: Charm++> cpu topology info is gathered in 0.001 seconds.
47: [0] No TreeLB configuration file found. Choosing a default configuration.
47: [0] TreeLB: Using PE_Root tree with: GreedyRefine 
47: 0 00000.00 Parameters  Adapt:max_level = 0
47: 0 00000.00 Parameters  Adapt:min_level = 0
47: 0 00000.00 Parameters  Boundary:type = "periodic"
47: 0 00000.00 Parameters  Domain:lower = [ -2.000000000000000 -2.000000000000000 -2.000000000000000 ]
47: 0 00000.00 Parameters  Domain:rank = 3
47: 0 00000.00 Parameters  Domain:upper = [ 2.000000000000000 2.000000000000000 2.000000000000000 ]
47: 0 00000.00 Parameters  Field:alignment = 8
47: 0 00000.00 Parameters  Field:courant = 0.3000000000000000
47: 0 00000.00 Parameters  Field:diffusion = true
47: 0 00000.00 Parameters  Field:dual_energy = true
47: 0 00000.00 Parameters  Field:gamma = 1.000000100000000
47: 0 00000.00 Parameters  Field:ghost_depth = 4
47: 0 00000.00 Parameters  Field:history = 1
47: 0 00000.00 Parameters  Field:list = [ "density" "velocity_x" "velocity_y" "velocity_z" "acceleration_x" "acceleration_y" "acceleration_z" "total_energy" "internal_energy" "pressure" "density_total" "density_particle" "density_particle_accumulate" "potential" "density_gas" "X" "B" "X_copy" "B_copy" "potential_copy" "diagonal" ]
47: 0 00000.00 Parameters  Field:padding = 0
47: 0 00000.00 Parameters  Initial:list = [ "merge_sinks_test" ]
47: 0 00000.00 Parameters  Initial:merge_sinks_test:particle_data_filename = "particles.dat"
47: 0 00000.00 Parameters  Mesh:root_blocks = [ 4 4 4 ]
47: 0 00000.00 Parameters  Mesh:root_rank = 3
47: 0 00000.00 Parameters  Mesh:root_size = [ 64 64 64 ]
47: 0 00000.00 Parameters  Method:list = [ "pm_update" "merge_sinks" ]
47: 0 00000.00 Parameters  Method:merge_sinks:merging_radius_cells = 0.1000000000000000
47: 0 00000.00 Parameters  Method:pm_update:max_dt = 0.01000000000000000
47: 0 00000.00 Parameters  Output:data:dir = [ "./Dir_%04d" "cycle" ]
47: 0 00000.00 Parameters  Output:data:field_list = [ "acceleration_x" "acceleration_y" "acceleration_z" "density_total" "potential" ]
47: 0 00000.00 Parameters  Output:data:name = [ "data-%04d-%04d.h5" "cycle" "proc" ]
47: 0 00000.00 Parameters  Output:data:particle_list = [ "sink" ]
47: 0 00000.00 Parameters  Output:data:schedule:step = 10
47: 0 00000.00 Parameters  Output:data:schedule:var = "cycle"
47: 0 00000.00 Parameters  Output:data:type = "data"
47: 0 00000.00 Parameters  Output:list = [ "data" ]
47: 0 00000.00 Parameters  Particle:batch_size = 4096
47: 0 00000.00 Parameters  Particle:list = [ "sink" ]
47: 0 00000.00 Parameters  Particle:mass_is_mass = "dummy"
47: 0 00000.00 Parameters  Particle:sink:attributes = [ "x" "default" "y" "default" "z" "default" "vx" "default" "vy" "default" "vz" "default" "ax" "default" "ay" "default" "az" "default" "mass" "default" "lifetime" "default" "creation_time" "default" "metal_fraction" "default" "is_copy" "int64" "id" "int64" ]
47: 0 00000.00 Parameters  Particle:sink:position = [ "x" "y" "z" ]
47: 0 00000.00 Parameters  Particle:sink:velocity = [ "vx" "vy" "vz" ]
47: 0 00000.00 Parameters  Stopping:time = 1.000000000000000
47: UNIT TEST BEGIN
47: 0 00000.09  ==============================================
47: 0 00000.09   
47: 0 00000.09    .oooooo.             oooo  oooo            
47: 0 00000.09   d8P'  `Y8b            `888  `888            
47: 0 00000.09  888           .ooooo.   888   888   .ooooo.  
47: 0 00000.09  888          d88' `88b  888   888  d88' `88b 
47: 0 00000.09  888          888ooo888  888   888  888   888 
47: 0 00000.09  `88b    ooo  888    .o  888   888  888   888 
47: 0 00000.09   `Y8bood8P'  `Y8bod8P' o888o o888o `Y8bod8P' 
47: 0 00000.09   
47: 0 00000.09  A Parallel Adaptive Mesh Refinement Framework
47: 0 00000.09   
47: 0 00000.09    Laboratory for Computational Astrophysics
47: 0 00000.09          San Diego Supercomputer Center
47: 0 00000.09       University of California, San Diego
47: 0 00000.09   
47: 0 00000.09  See 'LICENSE_CELLO' for software license information
47: 0 00000.09   
47: 0 00000.09  BEGIN CELLO: Jun 02 03:10:10
47: 0 00000.09 Define Simulation processors 4
47: 0 00000.09 Define CELLO_PREC          double
47: 0 00000.09 Define CC                  /nasa/intel/Compiler/2020.4.304/compilers_and_libraries_2020.4.304/linux/bin/intel64/icc
47: 0 00000.09 Define CFLAGS              
47: 0 00000.09 Define CPPDEFINES          CONFIG_USE_GRACKLE;BOOST_SYSTEM_NO_DEPRECATED;NO_FREETYPE;CONFIG_PRECISION_DOUBLE;SMALL_INTS;BYPASS_CHARM_MEM_LEAK;CONFIG_NODE_SIZE=64;CONFIG_NODE_SIZE_3=192;CONFIG_USE_PERFORMANCE;CONFIG_HAVE_VERSION_CONTROL
47: 0 00000.09 Define CXX                 /nasa/intel/Compiler/2020.4.304/compilers_and_libraries_2020.4.304/linux/bin/intel64/icpc
47: 0 00000.09 Define CXXFLAGS            
47: 0 00000.09 Define FORTRANFLAGS        -nofor-main -real-size 64 -double-size 64
47: 0 00000.09 Define FORTRAN             /nasa/intel/Compiler/2020.4.304/compilers_and_libraries_2020.4.304/linux/bin/intel64/ifort
47: 0 00000.09 Define BUILD HOST          Linux-3.10.0-1160.59.1.1chaos.ch6.x86_64
47: 0 00000.09 Define BUILD DIR           /nobackup/pgrete/src/enzo-e/build-icc-mpi-latest-boost
47: 0 00000.09 Define CHANGESET           44250615
47: 0 00000.09 Define CHARM_PATH          /nobackup/pgrete/src/charm/build-icc-mpi-latest
47: 0 00000.09 Define CONFIG_NODE_SIZE    64
47: 0 00000.09 Define CONFIG_SMP_MODE     no
47: 0 00000.09 Define BYPASS_CHARM_MEM_LEAK Yes
47: 0 00000.09 CHARM CkNumPes()           4
47: 0 00000.09 CHARM CkNumNodes()         4
47: 0 00000.09  BEGIN ENZO-E
47: 0 00000.09 Memory bytes 0 bytes_high 0
47: 0 00000.14 Output writing data file ././Dir_0000/data-0000-0000.h5
47: ------------- Processor 0 Exiting: Caught Signal ------------
47: Reason: Segmentation fault
47: [0] Stack Traceback:
47:   [0:0] enzo-e 0xa4060e CmiTimerIsSynchronized
47:   [0:1] libpthread.so.0 0x2aaaac9a4630 
47:   [0:2] enzo-e 0x850191 Param::value_to_string[abi:cxx11](int)
47:   [0:3] enzo-e 0x84f807 Param::write(_IO_FILE*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int)
47:   [0:4] enzo-e 0x8350cf Parameters::write(_IO_FILE*, int)
47:   [0:5] enzo-e 0x834763 Parameters::write(char const*, int)
47:   [0:6] enzo-e 0x89eef9 OutputData::write_block(Block const*)
47:   [0:7] enzo-e 0x7ffd02 Block::p_output_write(int, int)
47:   [0:8] enzo-e 0x7c422b CkIndex_Block::_call_p_output_write_marshall13(void*, void*)
47:   [0:9] enzo-e 0x9c3d01 CkLocRec::invokeEntry(CkMigratable*, void*, int, bool)
47:   [0:10] enzo-e 0x98e713 CkArrayBroadcaster::deliver(CkArrayMessage*, ArrayElement*, bool)
47:   [0:11] enzo-e 0x99b936 CkArray::recvBroadcast(CkMessage*)
47:   [0:12] enzo-e 0x98490c CkDeliverMessageFree
47:   [0:13] enzo-e 0x98663e 
47:   [0:14] enzo-e 0x9888bf _processHandler(void*, CkCoreState*)
47:   [0:15] enzo-e 0xa1ca45 CsdScheduleForever
47:   [0:16] enzo-e 0xa1cc82 CsdScheduler
47:   [0:17] enzo-e 0xa40547 ConverseInit
47:   [0:18] enzo-e 0xa0dca2 charm_main
47:   [0:19] libc.so.6 0x2aaaad5e0555 __libc_start_main
47:   [0:20] enzo-e 0x56440f 
47: MPT ERROR: Rank 0(g:0) is aborting with error code 1.
47: 	Process ID: 51718, Host: r617i2n15, Program: /nobackupp2/pgrete/src/enzo-e/build-icc-mpi-latest-boost/bin/enzo-e
47: 	MPT Version: HPE MPT 2.23  11/19/20 04:21:56
47: 
47: MPT: --------stack traceback-------
47: MPT: Attaching to program: /proc/51718/exe, process 51718
47: MPT: [Thread debugging using libthread_db enabled]
47: MPT: Using host libthread_db library "/lib64/libthread_db.so.1".
47: MPT: (no debugging symbols found)...done.
47: MPT: (no debugging symbols found)...done.
47: MPT: (no debugging symbols found)...done.
47: MPT: (no debugging symbols found)...done.
47: MPT: (no debugging symbols found)...done.
47: MPT: (no debugging symbols found)...done.
47: MPT: 0x00002aaaac9a418c in waitpid () from /lib64/libpthread.so.0
47: MPT: warning: File "/nasa/pkgsrc/toss3/2020Q4/gcc10/lib64/libstdc++.so.6.0.28-gdb.py" auto-loading has been declined by your `auto-load safe-path' set to "$debugdir:$datadir/auto-load:/usr/bin/mono-gdb.py".
47: MPT: To enable execution of this file add
47: MPT: 	add-auto-load-safe-path /nasa/pkgsrc/toss3/2020Q4/gcc10/lib64/libstdc++.so.6.0.28-gdb.py
47: MPT: line to your configuration file "/home1/pgrete/.gdbinit".
47: MPT: To completely disable this security protection add
47: MPT: 	set auto-load safe-path /
47: MPT: line to your configuration file "/home1/pgrete/.gdbinit".
47: MPT: For more information about this security protection see the
47: MPT: "Auto-loading safe path" section in the GDB manual.  E.g., run from the shell:
47: MPT: 	info "(gdb)Auto-loading safe path"
47: MPT: Missing separate debuginfos, use: debuginfo-install glibc-2.17-325.el7_9.x86_64 libibverbs-41mlnx1-OFED.4.9.3.0.0.49417.x86_64 libmlx4-41mlnx1-OFED.4.7.3.0.3.50218.x86_64 libmlx5-41mlnx1-OFED.5.0.0.4.0.50218.x86_64 libnl3-3.2.28-4.el7.x86_64 libnuma-3.0sgi-sgi716r61.rhel73.x86_64
47: MPT: (gdb) #0  0x00002aaaac9a418c in waitpid () from /lib64/libpthread.so.0
47: MPT: #1  0x00002aaaacce7826 in mpi_sgi_system (
47: MPT: #2  MPI_SGI_stacktraceback (
47: MPT:     header=header@entry=0x7fffffff8d40 "MPT ERROR: Rank 0(g:0) is aborting with error code 1.\n\tProcess ID: 51718, Host: r617i2n15, Program: /nobackupp2/pgrete/src/enzo-e/build-icc-mpi-latest-boost/bin/enzo-e\n\tMPT Version: HPE MPT 2.23  11/1"...) at sig.c:340
47: MPT: #3  0x00002aaaacc19db9 in print_traceback (ecode=ecode@entry=1) at abort.c:246
47: MPT: #4  0x00002aaaacc1a08a in PMPI_Abort (comm=<optimized out>, errorcode=1)
47: MPT:     at abort.c:68
47: MPT: #5  0x0000000000a3debb in LrtsAbort(char const*) ()
47: MPT: #6  0x0000000000a3df23 in CmiAbortHelper(char const*, char const*, char const*, int, int) ()
47: MPT: #7  0x0000000000a4060e in KillOnAllSigs(int) ()
47: MPT: #8  <signal handler called>
47: MPT: #9  0x0000000000850191 in Param::value_to_string[abi:cxx11](int) ()
47: MPT: #10 0x000000000084f807 in Param::write(_IO_FILE*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int) ()
47: MPT: #11 0x00000000008350cf in Parameters::write(_IO_FILE*, int) ()
47: MPT: #12 0x0000000000834763 in Parameters::write(char const*, int) ()
47: MPT: #13 0x000000000089eef9 in OutputData::write_block(Block const*) ()
47: MPT: #14 0x00000000007ffd02 in Block::p_output_write(int, int) ()
47: MPT: #15 0x00000000007c422b in CkIndex_Block::_call_p_output_write_marshall13(void*, void*) ()
47: MPT: #16 0x00000000009c3d01 in CkLocRec::invokeEntry(CkMigratable*, void*, int, bool) ()
47: MPT: #17 0x000000000098e713 in CkArrayBroadcaster::deliver(CkArrayMessage*, ArrayElement*, bool) ()
47: MPT: #18 0x000000000099b936 in CkArray::recvBroadcast(CkMessage*) ()
47: MPT: #19 0x000000000098490c in CkDeliverMessageFree ()
47: MPT: #20 0x000000000098663e in _deliverForBocMsg(CkCoreState*, int, envelope*, IrrGroup*) [clone .constprop.0] ()
47: MPT: #21 0x00000000009888bf in _processHandler(void*, CkCoreState*) ()
47: MPT: #22 0x0000000000a1ca45 in CsdScheduleForever ()
47: MPT: #23 0x0000000000a1cc82 in CsdScheduler ()
47: MPT: #24 0x0000000000a40547 in ConverseInit ()
47: MPT: #25 0x0000000000a0dca2 in charm_main ()
47: MPT: #26 0x00002aaaad5e0555 in __libc_start_main () from /lib64/libc.so.6
47: MPT: #27 0x000000000056440f in _start ()
47: MPT: (gdb) A debugging session is active.
47: MPT: 
47: MPT: 	Inferior 1 [process 51718] will be detached.
47: MPT: 
47: MPT: Quit anyway? (y or n) [answered Y; input not from terminal]
47: MPT: Detaching from program: /proc/51718/exe, process 51718
47: MPT: [Inferior 1 (process 51718) detached]
47: 
47: MPT: -----stack traceback ends-----
47: MPT ERROR: MPI_COMM_WORLD rank 0 has terminated without calling MPI_Finalize()
47: 	aborting job
47: /nobackup/pgrete/src/ytenv/lib/python3.8/site-packages/yt/utilities/logger.py:4: VisibleDeprecationWarning: The configuration file /home1/pgrete/.config/yt/ytrc is deprecated in favor of /home1/pgrete/.config/yt/yt.toml. Currently, both are present. Please manually remove the deprecated one to silence this warning.
47: Deprecated since v4.0.0. This feature will be removed in v4.1.0
47:   from yt.config import ytcfg
47: Traceback (most recent call last):
47:   File "input/merge_sinks/run_merge_sinks_test.py", line 108, in <module>
47:     make_images(input_prefix = "Dir", output_prefix = "image")
47:   File "/nobackupp2/pgrete/src/enzo-e/input/merge_sinks/images.py", line 26, in make_images
47:     for ds in yt.DatasetSeries(ds_pattern):
47:   File "/nobackup/pgrete/src/ytenv/lib/python3.8/site-packages/yt/data_objects/time_series.py", line 155, in __new__
47:     outputs = cls._get_filenames_from_glob_pattern(outputs)
47:   File "/nobackup/pgrete/src/ytenv/lib/python3.8/site-packages/yt/data_objects/time_series.py", line 204, in _get_filenames_from_glob_pattern
47:     raise FileNotFoundError(f"No match found for pattern : {pattern}")
47: FileNotFoundError: No match found for pattern : Dir_????/Dir_????.block_list
1/1 Test #47: merge_sinks_stationary_parallel ...***Failed    9.62 sec

0% tests passed, 1 tests failed out of 1

Label Time Summary:
enzo          =   9.62 sec*proc (1 test)
parallel      =   9.62 sec*proc (1 test)
regression    =   9.62 sec*proc (1 test)

Total Test time (real) =   9.63 sec

The following tests FAILED:
	 47 - merge_sinks_stationary_parallel (Failed)
Errors while running CTest

```

**Comment by stefanarridge** [comment]
So looks like something goes wrong when writing parameters... I wonder if it has something to do with the `Particle:mass_is_mass` parameter. It's strange that this only seems to go wrong for parallel runs.

**Comment by tumlinson** [comment]
This output file is an accurate reflection of what I got when I ran this test.

**Comment by stefanarridge** [comment]
Can you checkout the `no-mass-is-mass` branch from my fork (https://github.com/stefanarridge/enzo-e/tree/no-mass-is-mass), and then build and run the tests again, and let me know what happens?

**Comment by tumlinson** [comment]
Hi @stefanarridge the no-mass-is-mass branch builds and runs the `merge_sinks_drift_serial` and `merge_sinks_drift_parallel` tests PASS. Many of the other routine tests fail, perhaps this branch is out of sync some other way. But the two we were focused on, work out.

**Comment by tumlinson** [comment]
The remaining failures look like this: 
`PBS r585i5n3:/nobackup/jtumlins/no-mass/enzo-e/build-icc-mpi> ctest -R Bound-Outflow-3D -V 
UpdateCTestConfiguration  from :/nobackup/jtumlins/no-mass/enzo-e/build-icc-mpi/DartConfiguration.tcl
Parse Config file:/nobackup/jtumlins/no-mass/enzo-e/build-icc-mpi/DartConfiguration.tcl
UpdateCTestConfiguration  from :/nobackup/jtumlins/no-mass/enzo-e/build-icc-mpi/DartConfiguration.tcl
Parse Config file:/nobackup/jtumlins/no-mass/enzo-e/build-icc-mpi/DartConfiguration.tcl
Test project /nobackup/jtumlins/no-mass/enzo-e/build-icc-mpi
Constructing a list of tests
Done constructing a list of tests
Updating test list for fixtures
Added 0 tests to meet fixture requirements
Checking test dependency graph...
Checking test dependency graph end
test 11
    Start 11: Bound-Outflow-3D

11: Test command: /home1/jtumlins/anaconda3/bin/python "/nobackup/jtumlins/no-mass/enzo-e/tools/run_cpp_test.py" "--output-dump" "/nobackup/jtumlins/no-mass/enzo-e/build-icc-mpi/test/BoundaryConditions/Outflow-3D/Bound-Outflow-3D.log" "mpiexec" "-np" "4" "/nobackup/jtumlins/no-mass/enzo-e/build-icc-mpi/bin/enzo-e" "input/Boundary/boundary_outflow-3d.in"
11: Test timeout computed to be: 1500
11: usage: run_cpp_test.py [-h] [--output-dump OUTPUT_DUMP] COMMAND [ARGS ...]
11: run_cpp_test.py: error: unrecognized arguments: -np 4 /nobackup/jtumlins/no-mass/enzo-e/build-icc-mpi/bin/enzo-e input/Boundary/boundary_outflow-3d.in
1/1 Test #11: Bound-Outflow-3D .................***Failed    0.06 sec

0% tests passed, 1 tests failed out of 1

Label Time Summary:
enzo        =   0.06 sec*proc (1 test)
parallel    =   0.06 sec*proc (1 test)

Total Test time (real) =   0.06 sec

The following tests FAILED:
	 11 - Bound-Outflow-3D (Failed)
Errors while running CTest`

but again, these tests pass on `main`

**Comment by stefanarridge** [comment]
> but again, these tests pass on `main`

Is that the latest version of `main`? I ask because it looks like the errors are due to changes introduced by PR #192 (@mabruzzo), which was merged into 'main' yesterday.

Regarding the `mass-is-mass` parameter issue, perhaps it is best that I remove it. It would mean that users would have to include that parameter in the input file in order for yt to read in the particle masses correctly. Going forward, it would probably be better to use a version numbering system, maybe something for @pgrete and @brittonsmith to discuss.

**Comment by tumlinson** [comment]
@stefanarridge you're right. When I said these tests pass on `main` I was referring to my earlier tests on PR139, before #192 was merged. Now that I pull `main` right now, many of the "easy" tests fail with the same error. 

in fact, the apparent cause is something that has bedeviled me throughout my recent testing. all pathnames to input files for automatic tests and example problems usually start with "input/" but that assumes enzo-e is being run in ~/src/enzo-e or whatever the main directory for the repo is. This is usually not true, and in this case 'make test' has to be run from build-icc-mpi, which makes the relative path fed into `run_cpp_test.py` incorrect. See error trace below. It would be good to have a more elegant way to deal with these path assumptions. 
 
cc @mabruzzo 

---------------------------------

jtumlins@pfe22:/nobackup/jtumlins/no-mass/enzo-e/build-icc-mpi> ctest -R AmrPpm -V > AmrPpm.out 2>&1
jtumlins@pfe22:/nobackup/jtumlins/no-mass/enzo-e/build-icc-mpi> more AmrPpm.out 
UpdateCTestConfiguration  from :/nobackup/jtumlins/no-mass/enzo-e/build-icc-mpi/DartConfiguration.tcl
Parse Config file:/nobackup/jtumlins/no-mass/enzo-e/build-icc-mpi/DartConfiguration.tcl
UpdateCTestConfiguration  from :/nobackup/jtumlins/no-mass/enzo-e/build-icc-mpi/DartConfiguration.tcl
Parse Config file:/nobackup/jtumlins/no-mass/enzo-e/build-icc-mpi/DartConfiguration.tcl
Test project /nobackup/jtumlins/no-mass/enzo-e/build-icc-mpi
Constructing a list of tests
Done constructing a list of tests
Updating test list for fixtures
Added 0 tests to meet fixture requirements
Checking test dependency graph...
Checking test dependency graph end
test 5
    Start 5: AmrPpm

5: Test command: /home1/jtumlins/anaconda3/bin/python "/nobackup/jtumlins/no-mass/enzo-e/tools/run_cpp_test.py" "--output-dump" "/nobackup/jtumlins/no-mass/enzo-e/build-icc-mpi/test/Am
rPpm/Adapt/AmrPpm.log" "mpiexec" "-np" "4" "/nobackup/jtumlins/no-mass/enzo-e/build-icc-mpi/bin/enzo-e" "input/Adapt/adapt-L5-P1.in"
5: Test timeout computed to be: 1500
5: usage: run_cpp_test.py [-h] [--output-dump OUTPUT_DUMP] COMMAND [ARGS ...]
5: run_cpp_test.py: error: unrecognized arguments: -np 4 /nobackup/jtumlins/no-mass/enzo-e/build-icc-mpi/bin/enzo-e input/Adapt/adapt-L5-P1.in
1/1 Test #5: AmrPpm ...........................***Failed    0.08 sec

0% tests passed, 1 tests failed out of 1

Label Time Summary:
enzo        =   0.08 sec*proc (1 test)
parallel    =   0.08 sec*proc (1 test)

Total Test time (real) =   0.09 sec

The following tests FAILED:
	  5 - AmrPpm (Failed)
Errors while running CTest

**Comment by tumlinson** [comment]
However it is not just the pathnames. @mabruzzo the "unrecognized arguments" error persists even if I set the test input file paths correctly in test/CTestTestfile.cmake. 

Since #192 was closed and merged I am not sure what to do here.

**Comment by gregbryan** [comment]
Weird because it works for me and for circleci. I'm wondering if this is some weird behavior with python argparse -- in particular, I am thinking that run_cpp_test.py is interpreting `-np` at a new argument rather as one of the arguments to pass along to COMMAND.  @tumlinson: Can you test one thing for me?  In run_cpp_test.py can you change line 29 from:
`"args_for_command", metavar = "ARGS", action = "store", nargs = '*',`
to:
`"args_for_command", metavar = "ARGS", action = "store", nargs = argparse.REMAINDER,`
and try that?

(Agreed that the filename issue could be dealt with better but that is a different issue)

**Comment by tumlinson** [comment]
@gregbryan that did the trick. 

So, update. Here is the test record for `main` as of right now - cloned this morning. 

only one failure, which is for vlct_MHD_shock_test: 

41: FAILED: Standard Deviation of norm of L1 error of y-axis rj2a shock tube N=256 isn't correct
41: computed: 7.882023029284987e-17 expected: 0.0
41: FAILED: Standard Deviation of norm of L1 error of z-axis rj2a shock tube N=256 isn't correct
41: computed: 3.09710088675036e-17 expected: 0.0

----------------------------



- Running tests...
Test project /nobackup/jtumlins/no-mass/enzo-e/build-icc-mpi
      Start  1: Array
 1/49 Test  #1: Array .............................   Passed    0.32 sec
      Start  2: Error
 2/49 Test  #2: Error .............................   Passed    0.16 sec
      Start  3: Memory
 3/49 Test  #3: Memory ............................   Passed    0.15 sec
      Start  4: Monitor
 4/49 Test  #4: Monitor ...........................   Passed    0.15 sec
      Start  5: AmrPpm
 5/49 Test  #5: AmrPpm ............................   Passed  204.14 sec
      Start  6: Bound-Reflect-2D
 6/49 Test  #6: Bound-Reflect-2D ..................   Passed    5.42 sec
      Start  7: Bound-Periodic-2D
 7/49 Test  #7: Bound-Periodic-2D .................   Passed    5.56 sec
      Start  8: Bound-Outflow-2D
 8/49 Test  #8: Bound-Outflow-2D ..................   Passed    5.35 sec
      Start  9: Bound-Reflect-3D
 9/49 Test  #9: Bound-Reflect-3D ..................   Passed    9.09 sec
      Start 10: Bound-Periodic-3D
10/49 Test #10: Bound-Periodic-3D .................   Passed    9.40 sec
      Start 11: Bound-Outflow-3D
11/49 Test #11: Bound-Outflow-3D ..................   Passed    9.06 sec
      Start 12: FluxCorrect-SMR-PPM
12/49 Test #12: FluxCorrect-SMR-PPM ...............   Passed   13.59 sec
      Start 13: FluxCorrect-SMR-VL
13/49 Test #13: FluxCorrect-SMR-VL ................   Passed    6.94 sec
      Start 14: GravityCg-1
14/49 Test #14: GravityCg-1 .......................   Passed   38.36 sec
      Start 15: GravityCg-8
15/49 Test #15: GravityCg-8 .......................   Passed   19.10 sec
      Start 16: Heat-1
16/49 Test #16: Heat-1 ............................   Passed   31.05 sec
      Start 17: Heat-8
17/49 Test #17: Heat-8 ............................   Passed   48.68 sec
      Start 18: Music-111
18/49 Test #18: Music-111 .........................   Passed    0.35 sec
      Start 19: Music-222
19/49 Test #19: Music-222 .........................   Passed    0.41 sec
      Start 20: Music-444
20/49 Test #20: Music-444 .........................   Passed    1.07 sec
      Start 21: Music-211
21/49 Test #21: Music-211 .........................   Passed    0.34 sec
      Start 22: Music-121
22/49 Test #22: Music-121 .........................   Passed    0.34 sec
      Start 23: Music-112
23/49 Test #23: Music-112 .........................   Passed    0.34 sec
      Start 24: Music-411
24/49 Test #24: Music-411 .........................   Passed    0.37 sec
      Start 25: Music-141
25/49 Test #25: Music-141 .........................   Passed    0.36 sec
      Start 26: Music-114
26/49 Test #26: Music-114 .........................   Passed    0.36 sec
      Start 27: Output-Stride-1
27/49 Test #27: Output-Stride-1 ...................   Passed    0.85 sec
      Start 28: Output-Stride-2
28/49 Test #28: Output-Stride-2 ...................   Passed    0.85 sec
      Start 29: Output-Stride-4
29/49 Test #29: Output-Stride-4 ...................   Passed    0.85 sec
      Start 30: Output-Headers
30/49 Test #30: Output-Headers ....................   Passed   14.50 sec
      Start 31: Particle-X
31/49 Test #31: Particle-X ........................   Passed    0.89 sec
      Start 32: Particle-Y
32/49 Test #32: Particle-Y ........................   Passed    0.88 sec
      Start 33: Particle-XY
33/49 Test #33: Particle-XY .......................   Passed    0.88 sec
      Start 34: Particle-Circle
34/49 Test #34: Particle-Circle ...................   Passed    7.46 sec
      Start 35: Particle-AMR-Static
35/49 Test #35: Particle-AMR-Static ...............   Passed    6.69 sec
      Start 36: Particle-AMR-Dynamic
36/49 Test #36: Particle-AMR-Dynamic ..............   Passed    7.60 sec
      Start 37: Performance-Initial-PNG
37/49 Test #37: Performance-Initial-PNG ...........   Passed    3.69 sec
      Start 38: PPM-1
38/49 Test #38: PPM-1 .............................   Passed    8.57 sec
      Start 39: PPM-8
39/49 Test #39: PPM-8 .............................   Passed    5.45 sec
      Start 40: vlct_dual_energy_cloud
40/49 Test #40: vlct_dual_energy_cloud ............   Passed   12.30 sec
      Start 41: vlct_MHD_shock_tube
41/49 Test #41: vlct_MHD_shock_tube ...............***Failed   57.77 sec
      Start 42: vlct_MHD_linear_wave
42/49 Test #42: vlct_MHD_linear_wave ..............   Passed  144.28 sec
      Start 43: vlct_HD_linear_wave
43/49 Test #43: vlct_HD_linear_wave ...............   Passed   86.54 sec
      Start 44: vlct_passive_advect_sound
44/49 Test #44: vlct_passive_advect_sound .........   Passed   18.62 sec
      Start 45: vlct_dual_energy_shock_tube
45/49 Test #45: vlct_dual_energy_shock_tube .......   Passed  103.35 sec
      Start 46: merge_sinks_stationary_serial
46/49 Test #46: merge_sinks_stationary_serial .....   Passed   78.90 sec
      Start 47: merge_sinks_stationary_parallel
47/49 Test #47: merge_sinks_stationary_parallel ...   Passed   70.31 sec
      Start 48: merge_sinks_drift_serial
48/49 Test #48: merge_sinks_drift_serial ..........   Passed   74.70 sec
      Start 49: merge_sinks_drift_parallel
49/49 Test #49: merge_sinks_drift_parallel ........   Passed   66.14 sec

98% tests passed, 1 tests failed out of 49

Label Time Summary:
enzo          = 1181.79 sec*proc (45 tests)
parallel      = 606.22 sec*proc (24 tests)
regression    = 712.92 sec*proc (10 tests)
serial        = 576.36 sec*proc (25 tests)
unit          =   0.78 sec*proc (4 tests)

Total Test time (real) = 1182.69 sec

The following tests FAILED:
	 41 - vlct_MHD_shock_tube (Failed)
Errors while running CTest
make: *** [test] Error 8

**Comment by gregbryan** [comment]
Thanks Jason. I opened a PR for the run_cpp_test.py fix because that's easy and may affect others, depending on charm++ build. Regarding the VLCT error you are seeing, that looks a lot like round-off error but since it seems harmless I'll wait until Matthew comes back on-line (he is traveling right now).

**Comment by mabruzzo** [comment]
Sorry for not seeing this earlier today.

The remaining failure:
> 41: FAILED: Standard Deviation of norm of L1 error of y-axis rj2a shock tube N=256 isn't correct
> 41: computed: 7.882023029284987e-17 expected: 0.0
> 41: FAILED: Standard Deviation of norm of L1 error of z-axis rj2a shock tube N=256 isn't correct
> 41: computed: 3.09710088675036e-17 expected: 0.0

is worrisome. This is indicative of an underlying issue in the VL+CT solver. Basically this test problem runs a bunch of identical 1D shock tube setups in parallel. The shock tubes should all look exactly the same at the final time (the standard deviation should be 0).

I don't suppose you have turned on floating-point value unsafe optimizations (like `-ffast-math`)? If not, this suggests a minor issue related to constrained-transport 

(But, you don't need to worry about it right now while running pure-hydro cosmology simulations).

**Comment by mabruzzo** [comment]
Ah, the issue is that you are using the intel compilers! In comparison to other compilers like clang or gcc where you have to explicitly opt into value-unsafe floating point optimzations (like reordering `a+b+c` as `b+a+c` or replacing `a*b + a*c` with `a*(b+c)`), the intel compilers enables these optimizations by default. This is the main reason that intel compilers has a reputation for being faster than other compilers. (the floating point optimizations make auto-vectorization faster)

I definitely wouldn't worry about this failing test. I should update the test error message to warn people about this. However, if you're really concerned about this, I've provided instructions in the following spoiler tag that should confirm this isn't a problem:

<details> 
  <summary> Instructions to disable intel value-unsafe floating-point optimizations </summary>

  If you are really concerned about this, you would need to make a minor change in `config/pleiades_icc.cmake`. Replace the line:
  ```
    set(CMAKE_Fortran_FLAGS "-nofor-main" CACHE STRING "Default Fortran flags")
  ```
  with:
  ```
    set(CMAKE_Fortran_FLAGS "-nofor-main" CACHE STRING "Default Fortran flags")
    set(CMAKE_C_FLAGS "-fp-model precise" CACHE STRING "Default C flags")
    set(CMAKE_CXX_FLAGS "${CMAKE_C_FLAGS}" CACHE STRING "Default C++ flags")
  ```
</details>


> 10^-17 does seem small, but it isn't zero. . .

A feature of the VL+CT solver is that does a pretty good job preserving symmetry (it's not perfect - to be perfect you need to start making choices that slow things down). This feature is extremely useful during code development (i.e. If a non-zero standard deviation creeps in here, that's a sign that a code-change broke something).

With that said, I'm totally on the same page as you for production runs (where you don't care about symmetry). I'm all for using these value-unsafe optimizations to speed up the code in this case. (In fact, if you append `-DUSE_SIMD=ON` that should facilitate extra floating-point optimizations that will significantly speed up the VL+CT solver in pure-hydro mode - note this hasn't been tested this on Pleiades).

**Comment by tumlinson** [comment]
Thanks for that thorough explanation @mabruzzo. In my role as Pleiades test particle I am wondering if we should fine tune the default configuration for Pleiades as you suggest (in your spoiler) so that the config passes all tests, rather than leaving one to fail unless a change is made. OTOH I think we would not want to have a default config that is not well optimized. I guess this is a philosophical point. 

Nevertheless from the point of view of this particular issue on the merge_sinks test, it's fixed and we can close the issue? Right @stefanarridge ?

**Comment by stefanarridge** [comment]
I'd say it will be safe to close the issue when PR #193 is merged into `main`.
